rust/relay/src/server.rs

```diff
@@ -58,6 +58,10 @@ pub struct Server<R> {
     clients_by_allocation: HashMap<AllocationId, SocketAddr>,
     allocations_by_port: HashMap<u16, AllocationId>,
 
+    lowest_port: u16,
+    highest_port: u16,
+    max_available_ports: u16,
```

Comment:
I'd recommend to make this a private function of the server to make sure it is always computed from the two other fields.

Reply:
Good call. Updated